auth/src/main/java/com/farsitel/bazaar/storage/connection/ReceiverStorageConnection.kt

```diff
@@ -0,0 +1,154 @@
+package com.farsitel.bazaar.storage.connection
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.Observer
+import com.farsitel.bazaar.BAZAAR_PACKAGE_NAME
+import com.farsitel.bazaar.storage.callback.BazaarStorageCallback
+import com.farsitel.bazaar.storage.receiver.StorageReceiver
+import com.farsitel.bazaar.util.AbortableCountDownLatch
+import com.farsitel.bazaar.util.InAppLoginLogger
+import com.farsitel.bazaar.util.fromBase64
+import com.farsitel.bazaar.util.ext.toBase64
+
+internal class ReceiverStorageConnection(
+    private val context: Context
+) : StorageConnection(context) {
+
+    private var bazaarSetStorageCallback: BazaarStorageCallback? = null
+    private var bazaarGetStorageCallback: BazaarStorageCallback? = null
+
+    private var getStorageLatch: AbortableCountDownLatch? = null
+    private var setStorageLatch: AbortableCountDownLatch? = null
+
+    private var appPayload: String? = null
+
+    private val observer = Observer<Intent> { intent ->
+        when (intent.action) {
+            ACTION_STORAGE_GET_DATA_RESPONSE -> {
+                handleGetSavedDataResponse(intent.extras)
+            }
+            ACTION_STORAGE_SET_DATA_RESPONSE -> {
+                handleSetDataResponse(intent.extras)
+            }
+            ACTION_STORAGE_SET_DATA,
+            ACTION_STORAGE_GET_DATA
+            -> {
+                context.sendBroadcast(
+                    getNewIntentForBroadcast(
+                        requireNotNull(intent.action),
+                        intent.extras
+                    )
+                )
+            }
```

Comment:
```suggestion
            ACTION_STORAGE_SET_DATA,
            ACTION_STORAGE_GET_DATA -> {
                context.sendBroadcast(
                    getNewIntentForBroadcast(
                        requireNotNull(intent.action),
                        intent.extras
                    )
                )
            }
```
This looks better IMO.

Reply:
IDE suggestion